### ISSUE TITLE
Related to spook?

### ISSUE THREAD

**Issue by emufan** [frenck/spook#183]
Ofc I don't use spook as is is not allowed or recommended. 😉 But let's assume I would and would find these lines in my log often but now and then directly after I would have installed it:

```
Error doing job: Task exception was never retrieved
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/util/logging.py", line 185, in coro_wrapper
    return await target
TypeError: object NoneType can't be used in 'await' expression

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/util/logging.py", line 187, in coro_wrapper
    log_exception(format_err, *args)
  File "/usr/src/homeassistant/homeassistant/util/logging.py", line 115, in log_exception
    friendly_msg = format_err(*args)
  File "/usr/src/homeassistant/homeassistant/util/logging.py", line 207, in <lambda>
    target.__name__,
AttributeError: 'NoneType' object has no attribute '__name__'
```

Would it be possible that it is related to spook?

**Comment by frenck** [comment]
I can't say, as it doesn't reflect any of that from the shared information/logs.

../Frenck

**Comment by emufan** [comment]
I can't either. But this are the only messages several times in my log. It is starting after enabling and installing the integration and stopping directly after disabling the integration.

I would share more, if I would see anything more.

**Comment by frenck** [comment]
That is a helpful comment, thanks! 👍 

../Frenck

**Comment by emufan** [comment]
This is here still there 1:1 only when Spook is enabled. Updated to 1.1.0 today. Currently on 2023.08.2. Still there. When disabled it is never occurring.

New snapshow

```
[...]
2023-08-14 17:12:45.262 DEBUG (MainThread) [custom_components.spook] Registering Spook repairs: script.script_unknown_area_references
2023-08-14 17:12:45.268 DEBUG (MainThread) [custom_components.spook] Registering Spook repairs: script.script_unknown_device_references
2023-08-14 17:12:45.284 DEBUG (MainThread) [custom_components.spook] Registering Spook repairs: integration.integration_unknown_source
2023-08-14 17:12:45.300 DEBUG (MainThread) [custom_components.spook] Registering Spook repairs: group.group_unknown_members
2023-08-14 17:12:45.331 DEBUG (MainThread) [custom_components.spook] Registering Spook repairs: switch_as_x.switch_as_x_unknown_source
2023-08-14 17:12:46.389 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 185, in coro_wrapper
return await target
^^^^^^^^^^^^
TypeError: object NoneType can't be used in 'await' expression
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 187, in coro_wrapper
log_exception(format_err, *args)
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 115, in log_exception
friendly_msg = format_err(*args)
^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 207, in <lambda>
target.__name__,
^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute '__name__'
2023-08-14 17:12:46.392 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 185, in coro_wrapper
return await target
^^^^^^^^^^^^
TypeError: object NoneType can't be used in 'await' expression
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 187, in coro_wrapper
log_exception(format_err, *args)
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 115, in log_exception
friendly_msg = format_err(*args)
^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 207, in <lambda>
target.__name__,
^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute '__name__'
2023-08-14 17:12:46.396 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 185, in coro_wrapper
return await target
^^^^^^^^^^^^
TypeError: object NoneType can't be used in 'await' expression
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 187, in coro_wrapper
log_exception(format_err, *args)
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 115, in log_exception
friendly_msg = format_err(*args)
^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 207, in <lambda>
target.__name__,
^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute '__name__'
2023-08-14 17:12:46.400 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 185, in coro_wrapper
return await target
^^^^^^^^^^^^
TypeError: object NoneType can't be used in 'await' expression
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 187, in coro_wrapper
log_exception(format_err, *args)
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 115, in log_exception
friendly_msg = format_err(*args)
^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 207, in <lambda>
target.__name__,
^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute '__name__'
2023-08-14 17:12:46.403 ERROR (MainThread) [homeassistant] Error doing job: Task exception was never retrieved
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 185, in coro_wrapper
return await target
^^^^^^^^^^^^
TypeError: object NoneType can't be used in 'await' expression
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 187, in coro_wrapper
log_exception(format_err, *args)
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 115, in log_exception
friendly_msg = format_err(*args)
^^^^^^^^^^^^^^^^^
File "/usr/src/homeassistant/homeassistant/util/logging.py", line 207, in <lambda>
target.__name__,
^^^^^^^^^^^^^^^
AttributeError: 'NoneType' object has no attribute '__name__'
2023-08-14 17:12:55.151 DEBUG (MainThread) [custom_components.spook] Spook is inspecting: lovelace_unknown_entity_references
2023-08-14 17:12:55.155 DEBUG (MainThread) [custom_components.spook] Config for dashboard lovelace not found, skipping
2023-08-14 17:12:55.155 ERROR (MainThread) [custom_components.spook] Unexpected exception from <function AbstractSpookRepair.async_activate.<locals>._async_inspect at 0xffff74683420>
Traceback (most recent call last):
File "/usr/src/homeassistant/homeassistant/helpers/debounce.py", line 116, in _handle_timer_finish
await task
File "/config/custom_components/spook/repairs.py", line 126, in _async_inspect
await self.async_inspect()
File "/config/custom_components/spook/ectoplasms/lovelace/repairs/unknown_entity_references.py", line 112, in async_inspect
for entity_id in self.__async_extract_entities(config)
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/spook/ectoplasms/lovelace/repairs/unknown_entity_references.py", line 159, in __async_extract_entities
entities.update(self.__async_extract_entities_from_badge(badge))
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
File "/config/custom_components/spook/ectoplasms/lovelace/repairs/unknown_entity_references.py", line 207, in __async_extract_entities_from_badge
return set(config["entities"])
^^^^^^^^^^^^^^^^^^^^^^^
TypeError: unhashable type: 'dict'
2023-08-14 17:12:55.208 DEBUG (MainThread) [custom_components.spook] Spook is inspecting: automation_unknown_entity_references
2023-08-14 17:12:55.226 DEBUG (MainThread) [custom_components.spook] Spook is inspecting: automation_unknown_area_references
2023-08-14 17:12:55.233 DEBUG (MainThread) [custom_components.spook] Spook is inspecting: automation_unknown_device_references
2023-08-14 17:12:55.257 DEBUG (MainThread) [custom_components.spook] Spook is inspecting: script_unknown_entity_references
[...]
```

Where the last error is separated from the problem and I opened another issue for that in parallel.

**Comment by ReneNulschDE** [comment]
Hi,

I have read the license :-)

I can reproduce this in my environment too, but only when the log-level for spook is "info or higher". It does not happen when log-level is debug. With log-level debug all looks good.

Exception: https://homeass.ist/?14a611168633485d#5LdTbK4hC7sKrpknA4TJkzyMcaia3Mw1ZJR2YRXa8qjT

- started in recovery mode --> no exception
- removed all custom components (incl. spook) --> no exception
- added only spook with default log-level critical --> exception
- added only spook with default log-level critical and spook level: info --> exception
- added only spook with default log-level critical and spook level: debug --> no exception

Environment:
## System Information

version | core-2024.2.0.dev0
-- | --
installation_type | Home Assistant Core
dev | true
hassio | false
docker | false
user | rene
virtualenv | true
python_version | 3.12.1
os_name | Linux
os_version | 6.5.0-15-generic
arch | x86_64
timezone | Europe/Berlin
config_dir | /home/rene/.homeassistant

**Comment by frenck** [comment]
I'm not able to reproduce any of this on the latest dev. There is quite some structural changes.

Let's close for now and see how it behaves from that point on.

../Frenck
